### ISSUE TITLE
Release code and models for ICoT on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [ajyl/icot#1]
Hi @ajyl 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2510.00184.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance), you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed that your GitHub repository (`https://github.com/ajyl/icot`) is present, but it appears the code and weights for your ICoT model and the auxiliary loss approach are not yet available. I also saw a comment on your paper page asking about training scripts, which suggests there's community interest in your implementation.

It would be great to make the code and any associated model checkpoints available on the 🤗 Hub, to improve their discoverability/visibility. We can add tags in the model cards so that people find the models easier, link them to the paper page, etc.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

Let me know if you're interested/need any help regarding this when you're ready to release!

Cheers,

Niels
ML Engineer @ HF 🤗
